### PR TITLE
[tests-only] add unit tests for the account page and minimize coverage from acceptance tests

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added unit tests for the account page
- removed accounts page related acceptance tests other than verifying the successful page load

## Related Issue

- Part of https://github.com/owncloud/web/issues/6475

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-runtime/src/pages/account.vue [hunk 1]
```diff
@@ -20,39 +32,35 @@
       <h2 v-translate class="oc-text-bold oc-text-initial oc-mb">Account Information</h2>
 
       <dl class="account-page-info oc-flex oc-flex-wrap">
-        <div class="oc-mb oc-width-1-2@s">
-          <dt v-translate class="account-page-info-username oc-text-normal oc-text-muted">
```
Comment (kiranparajuli589) [inline packages/web-runtime/src/pages/account.vue:24]: selectors are moved one level up (not only to the title of the account info" but wrapping whole div with `dt` and `dd` inside)

File: packages/web-runtime/src/pages/account.vue [hunk 2]
```diff
@@ -20,39 +32,35 @@
...
+        <div class="account-page-info-username oc-mb oc-width-1-2@s">
+          <dt v-translate class="oc-text-normal oc-text-muted">Username</dt>
           <dd>
             {{ user.username || user.id }}
           </dd>
         </div>
-        <div v-if="user.username && user.id">
-          <dt v-translate class="account-page-info-userid oc-text-normal oc-text-muted">User ID</dt>
+        <div v-if="user.username && user.id" class="account-page-info-userid">
+          <dt v-translate class="oc-text-normal oc-text-muted">User ID</dt>
```
Comment (kiranparajuli589) [inline packages/web-runtime/src/pages/account.vue:42]: selectors are moved one level up (not only to the title of the account info" but wrapping the whole div with `dt` and `dd` inside)

**Comment by ownclouders** [comment]
Results for <strong>oC10Basic1</strong> https://drone.owncloud.com/owncloud/web/23257/12/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIAccount-accountInformation_feature-L10.png
  

### webUIAccount-accountInformation_feature-L20.png
  

### webUIAccount-accountInformation_feature-L34.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermission</strong> https://drone.owncloud.com/owncloud/web/23282/34/1
The following scenarios passed on retry:
- webUISharingPermissionsUsers/sharePermissionsUsers.feature:122

**Review by saw-jan — COMMENTED** [review]
LGTM

File: packages/web-runtime/tests/unit/pages/account.spec.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  loaderStub: 'oc-loader-stub',
+  accountPageTitle: '#account-page-title',
+  editUrlButton: '[data-testid="account-page-edit-url-btn"]',
+  editRouteButton: '[data-testid="account-page-edit-route-btn"]',
+  accountPageInfo: '.account-page-info',
+  groupInformation: '.account-page-info-groups'
+}
+
+describe('account', () => {
+  describe('when the wrapper is still loading', () => {
```
Comment (kulmann) [inline packages/web-runtime/tests/unit/pages/account.spec.js:36]:
```suggestion
  describe('when the page is still loading', () => {
```

File: packages/web-runtime/tests/unit/pages/account.spec.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    })
+    it('should render the loading state', () => {
+      const pageTitle = wrapper.find(selectors.pageTitle)
+      const ocLoader = wrapper.find(selectors.loaderStub)
+      expect(pageTitle.text()).toBe($route.meta.title)
+      expect(ocLoader.exists()).toBeTruthy()
+    })
+    it('should not render the account page content', () => {
+      const accountPageTitle = wrapper.find(selectors.accountPageTitle)
+      const accountPageInfo = wrapper.find('.account-page-info')
```
Comment (kulmann) [inline packages/web-runtime/tests/unit/pages/account.spec.js:50]:
```suggestion
      const accountPageInfo = wrapper.find(selectors.accountPageInfo)
```

File: packages/web-runtime/tests/unit/pages/account.spec.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    })
+  })
+  describe('when the wrapper is not loading anymore', () => {
+    it('should not render the loading state', async () => {
+      const store = getStore()
+      const wrapper = getWrapper(store)
+      await wrapper.setData({ loading: false })
+      const pageTitle = wrapper.find(selectors.pageTitle)
+      const ocLoader = wrapper.find(selectors.loaderStub)
+      expect(pageTitle.text()).toBe('Account')
```
Comment (kulmann) [inline packages/web-runtime/tests/unit/pages/account.spec.js:62]: I'll allow it for this PR, but in general I don't like strings from the code hardcoded in tests. The implementation of the page title is wrong though, so let's keep it like it is, I'll adjust it in my PR mentioned before (if the code were correct you could use the title from your route mock, which makes the test code more resilient). Again, just FYI, nothing to do from your side.

File: packages/web-runtime/tests/unit/pages/account.spec.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+          const store = getStore({
+            server: 'http://server/address/',
+            isOcis: false
+          })
+          const wrapper = getWrapper(store)
+          await wrapper.setData({ loading: false })
+          const editUrlButton = wrapper.find(selectors.editUrlButton)
+          const editRouteButton = wrapper.find(selectors.editRouteButton)
+          expect(editUrlButton).toMatchSnapshot()
+          expect(editUrlButton.attributes()).toMatchSnapshot()
```
Comment (kulmann) [inline packages/web-runtime/tests/unit/pages/account.spec.js:78]: not needed as you already have a snapshot test for the whole button one line above.

File: packages/web-runtime/tests/unit/pages/account.spec.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    it('should render the loading state', () => {
+      const pageTitle = wrapper.find(selectors.pageTitle)
+      const ocLoader = wrapper.find(selectors.loaderStub)
+      expect(pageTitle.text()).toBe($route.meta.title)
+      expect(ocLoader.exists()).toBeTruthy()
+    })
+    it('should not render the account page content', () => {
+      const accountPageTitle = wrapper.find(selectors.accountPageTitle)
+      const accountPageInfo = wrapper.find('.account-page-info')
+      expect(accountPageTitle.exists()).toBeFalsy()
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-runtime/tests/unit/pages/account.spec.js:51]: happy that it helped :grin: :v:

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPerm3</strong> https://drone.owncloud.com/owncloud/web/23349/65/1
The following scenarios passed on retry:
- webUISharingFolderAdvancedPermissionMultipleUsers/sharedFolderWithMultipleUsersAdvancedPermissions.feature:60

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23349/67/1
The following scenarios passed on retry:
- webUISharingPublicExpire/shareByPublicLinkExpiringLinks.feature:11

**Pull request merged**